src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -583,6 +584,20 @@ ReactDOMComponent.Mixin = {
         );
         didWarnShadyDOM = true;
       }
+
+      if (__DEV__) {
+        if (this._namespaceURI === DOMNamespaces.html) {
+          warning(
+            !(ownerDocument.createElement(this._tag) instanceof window.HTMLUnknownElement) ||
+              isCustomComponent(this._tag, props) || voidElementTags[this._tag],
```

Comment:
It's not apparent to me why checking against `voidElementTags` makes sense here.